source/slang/slang-ir-redundancy-removal.cpp

```diff
@@ -145,6 +145,28 @@ bool removeRedundancyInFunc(IRGlobalValueWithCode* func)
     return result;
 }
 
+// Remove all [AvailableInDXIL] functions when compiling DXIL.
+void removeAvailableInDXIL(IRModule* module)
+{
+    List<IRInst*> toRemove;
+    for (auto globalInst : module->getGlobalInsts())
+    {
+        if (!as<IRFunc>(globalInst))
+        {
+            continue;
+        }
+        if (globalInst->findDecoration<IRAvailableInDXILDecoration>())
+        {
+            globalInst->findDecoration<IRNameHintDecoration>()->dump();
+            toRemove.add(globalInst);
+        }
+    }
+    for (auto inst : toRemove)
```

Comment:
I don't think you can remove the entire inst, because the glue HLSL still need to contain an extern declaration for the function. Also if you just remove the function inst, the IR will be in an invalid state because calls of those function will now be calls into wild pointers. You should just remove the body of the functions so it will be treated as a declaration in our emit logic.

Reply:
Could you recommend a test case that would show this problem?

Reply:
Maybe my test needs to call a normal export function from inside a generic function in the library module. Seems like that would show the problem since the module IR would have a reference to a function that gets deleted in this patch.

Reply:
Current revision guts the function instead of deleting it.